units/commit_queue.go

```diff
@@ -0,0 +1,352 @@
+	cancelCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	owner := projectRef.Owner
+	repo := projectRef.Repo
+	file := projectRef.CommitQConfigFile
+	remoteConfigFile, err := thirdparty.GetGithubFile(cancelCtx, githubToken, owner, repo, file, sha)
+	if err != nil {
+		return "", errors.Wrapf(err, "error fetching file %s from commit %s", file, sha)
+	}
+	configFileContents, err := base64.StdEncoding.DecodeString(*remoteConfigFile.Content)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to decode file %s", file)
+	}
+
+	return string(configFileContents), nil
+}
+
+func makeMergePatch(pr *github.PullRequest, projectID string, configString string) (*patch.Patch, error) {
+	u, err := getPatchUser(pr.GetUser().GetID())
+	if err != nil {
+		return nil, errors.Wrap(err, "can't get user for patch")
+	}
+	prAuthor := u.Id
+	patchNumber, err := u.IncPatchNumber()
+	if err != nil {
+		return nil, errors.Wrap(err, "error computing patch num")
+	}
+	id := bson.NewObjectId()
+
+	patchDoc := &patch.Patch{
+		Id:            id,
+		Project:       projectID,
+		Author:        prAuthor,
+		Githash:       *pr.Base.SHA,
+		Description:   fmt.Sprintf("Commit Queue merge test PR #%d", *pr.Number),
+		CreateTime:    time.Now(),
+		Status:        evergreen.PatchCreated,
+		PatchedConfig: configString,
+		PatchNumber:   patchNumber,
+		GithubPatchData: patch.GithubPatch{
+			PRNumber:       *pr.Number,
+			MergeCommitSHA: *pr.MergeCommitSHA,
+		},
+	}
+
+	return patchDoc, nil
+}
+
+func getPatchUser(userUID int) (*user.DBUser, error) {
+	u, err := user.FindByGithubUID(userUID)
+	if err != nil {
+		return nil, errors.Wrap(err, "can't look for user")
+	}
+	if u == nil {
+		// set to a default user
+		u, err = user.FindOne(user.ById(evergreen.GithubPatchUser))
+		if err != nil {
+			return nil, errors.Wrap(err, "can't get user for pull request")
+		}
+		// default user doesn't exist yet
+		if u == nil {
+			u = &user.DBUser{
+				Id:       evergreen.GithubPatchUser,
+				DispName: "Github Pull Requests",
+				APIKey:   util.RandomString(),
+			}
+			if err = u.Insert(); err != nil {
+				return nil, errors.Wrap(err, "failed to create github patch user")
+			}
+		}
+	}
+
+	return u, nil
+}
+
+func sendCommitQueueGithubStatus(pr *github.PullRequest, state message.GithubState, description, versionID string) error {
+	env := evergreen.GetEnvironment()
+	sender, err := env.GetSender(evergreen.SenderGithubStatus)
+	if err != nil {
+		return errors.Wrap(err, "can't get GitHub status sender")
+	}
+
+	url := ""
+	if versionID != "" {
+		url, _ = makeVersionURL(versionID)
+	}
+
+	msg := message.GithubStatus{
+		Owner:       *pr.Base.Repo.Owner.Login,
+		Repo:        *pr.Base.Repo.Name,
+		Ref:         *pr.Head.SHA,
+		Context:     "evergreen/commitqueue",
+		State:       state,
+		Description: "preparing to test merge",
```

Comment:
Shouldn't the description param be passed through here?

Reply:
definitely :flushed:
Done.